tests/test_documents.py

```diff
@@ -161,3 +166,101 @@ async def test_add_document_ingest_error(client):
         client=client, collection=collection.name, document_id=document.id
     )
     assert len(chunks) == 0
+
+async def test_upload_document_content_invalid(client, doc_fixture):
+    with open(NEARLY_EMPTY_PATH, 'rb') as file:
+        response = await upload_document_content.asyncio_detailed(
+            client=client,
+            document_id=1_000_000,
+            body=BodyUploadDocumentContent(
+                content=File(
+                    payload=file,
+                    file_name="file-name-1",
+                    mime_type="application/pdf",
+                ),
+            ),
+        )
+    assert response.status_code == 404
+
+async def test_document_lifecycle(client, doc_fixture):
+    # 1. Upload a PDF for one of the fixutre docs and verify the document is "pending"
+    with open(NEARLY_EMPTY_PATH, 'rb') as file:
```

Comment:
I pulled the content into `NEARLY_EMPTY_BYTES` so we don't need to reread the file every time.

Also, probably worth adding some functions for the two common things:

```
I thnik it's probably time we create a few helpers for readability. I'd propose:

```
async def test_update_document_content(client, doc_fixture):
  await upload_test_pdf(client, doc_fixture.doc1, NEARLY_EMPTY_PATH_BYTES)
  await document_ingested(client, doc_fixture.doc1)

...
```

Eg., the `upload_test_pdf` and `document_ingested` would be useful in a lot of tests, and would shorten this test and make the lifecycle more readable.